Improve the `PhysicsShapeQueryParameters3D`'s description

Yobbolita opened the pull request:
Made it clearer that this class is used in several of PhysicsDirectSpaceState3D's methods and not just on intersect_shape()

As you can see, this was looking kinda contradictory :

![question_godot_motion](https://github.com/user-attachments/assets/300fbf0d-fe26-425f-8d53-a3d6fde1ca3e)

<!--
Please target the `master` branch in priority.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.

To speed up the contribution process and avoid CI errors, please set up pre-commit hooks locally:
https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html
-->

Yobbolita committed:
Changed the wording on PhysicsShapeQueryParameters3D's description

Made it clearer that this class is used in several of PhysicsDirectSpaceState3D's methods and not just on intersect_shape()

Yobbolita committed:
Update PhysicsShapeQueryParameters3D.xml

Noted that the "motion" parameter is often unused to clarify the way the class works.

akien-mga commented on doc/classes/PhysicsShapeQueryParameters3D.xml:
```diff
@@ -1,7 +1,7 @@
+		Provides parameters for [PhysicsDirectSpaceState3D]'s methods.
 	</brief_description>
 	<description>
 		By changing various properties of this object, such as the shape, you can configure the parameters for [method PhysicsDirectSpaceState3D.intersect_shape].
```
This still refers to `intersect_shape` only.

Yobbolita replied:
I don't understand. I replaced it.

Yobbolita replied:
Done

akien-mga commented on doc/classes/PhysicsShapeQueryParameters3D.xml:
```diff
@@ -27,6 +27,7 @@
 		</member>
 		<member name="motion" type="Vector3" setter="set_motion" getter="get_motion" default="Vector3(0, 0, 0)">
 			The motion of the shape being queried for.
+			[b]Note:[/b] Many methods ignore this property.
```
That leaves the reader a bit hanging - how to know which methods ignore this property or not? Assuming this is properly documented in `PhysicsDirectSpaceState3D` method descriptions, I'd mention that here.
```suggestion
			[b]Note:[/b] Only some methods of [PhysicsDirectSpaceState3D] that take this class as parameter make use of this property. See their respective descriptions for details.
```

Mickeon commented:
Whatever is done should also be written in the 2D equivalent.

Yobbolita committed:
Added akien-mga's suggestion

Improved the "note" of the "motion" parameter

Co-authored-by: Rémi Verschelde <rverschelde@gmail.com>

Yobbolita commented:
> Whatever is done should also be written in the 2D equivalent.

It's true that the 2D version has the same class description problem

![screen16](https://github.com/user-attachments/assets/2ec3c4db-2b40-4046-bf4f-51a81b631f41)

However, none of PhysicsDirectSpaceState2D's methods mention that they ignore the "motion" parameter. So we would need to check if they are consistent with 3D in ignoring it or if they actually use it.

PhairZ commented on doc/classes/PhysicsShapeQueryParameters3D.xml:
```diff
@@ -27,6 +27,7 @@
 		</member>
 		<member name="motion" type="Vector3" setter="set_motion" getter="get_motion" default="Vector3(0, 0, 0)">
 			The motion of the shape being queried for.
+			[b]Note:[/b] Only some methods of [PhysicsDirectSpaceState3D] that take this class as parameter make use of this property. See their respective descriptions for details.
```
I don't think this is necessary to have.

Yobbolita replied:
Removed

PhairZ reviewed (commented):
This looks fine to me. I think the wording might be better but overall this is fine. Should be copied over to the 2D version of these.

PhairZ commented:
> However, none of PhysicsDirectSpaceState2D's methods mention that they ignore the "motion" parameter.

https://docs.godotengine.org/en/stable/classes/class_physicsdirectspacestate2d.html#class-physicsdirectspacestate2d-method-get-rest-info

Yobbolita commented:
Now what ? We wait for a "Godot admin" to come by ?

Yobbolita committed:
Changed a line that still referred to the intersect_shape method

Yobbolita committed:
Applied to the 2D class the same changes applied to the 3D class

Yobbolita commented:
I applied the changes to the 2D class as well. although we might still need to check which ones of the PhysicsDirectSpaceState2D methods actually use the motion parameter and which ones don't.

Yobbolita committed:
Removed note about "motion" parameter being often unused in 3d class

Mickeon commented on doc/classes/PhysicsShapeQueryParameters2D.xml:
```diff
@@ -30,6 +30,7 @@
 		</member>
 		<member name="shape" type="Resource" setter="set_shape" getter="get_shape">
 			The [Shape2D] that will be used for collision/intersection queries. This stores the actual reference which avoids the shape to be released while being used for queries, so always prefer using this over [member shape_rid].
+			[b]Note:[/b] Only some methods of [PhysicsDirectSpaceState2D] that take this class as parameter make use of this property. See their respective descriptions for details.
```
This could be deleted as well for similar reasons as the note in `motion` previously existing in this PR.
```suggestion
```

Yobbolita replied:
I think I intended on adding this note to the "motion" parameter rather than the "shape" parameter. Must have been tired.

Mickeon reviewed (commented):
This is overall a good change, I can say.

Ideally, you should be able to squash all of the commits into one. See the [Pull Request Workflow](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#the-interactive-rebase).

Yobbolita committed:
Removed note in PhysicsShapeQueryParameters2D.xml

Note wasn't needed as per PR comments

Note wasn't even in the correct plane to begin with (shape instead of motion)
